src/groupeddatatable/grouping.jl

```diff
@@ -371,12 +369,11 @@ end
 
 function _makeheaders{T<:Function}(fs::Vector{T}, cn::Vector{Symbol})
     fnames = _fnames(fs) # see other/utils.jl
-    reshape([Symbol(colname,'_',fname) for fname in fnames, colname in cn],
-            length(fnames)*length(cn))
+    [Symbol(colname,'_',fname) for fname in fnames for colname in cn]
 end
 
 function _aggregate{T<:Function}(d::AbstractDataTable, fs::Vector{T}, headers::Vector{Symbol}, sort::Bool=false)
-    res = DataTable(colwise(fs, d), headers)
+    res = DataTable(Any[vcat(f(d[i])) for f in fs for i in 1:size(d, 2)], headers)
```

Comment:
`vcat` does a nice job here since it will wrap `x` into `[x]` and will leave `[x]` as `[x]`. Splatting could also work, but `vec` doesn't in this case.

```julia
julia> vcat(1)
1-element Array{Int64,1}:
 1

julia> vcat([1])
1-element Array{Int64,1}:
 1

julia> vec([1])
1-element Array{Int64,1}:
 1

julia> vec(1)
ERROR: MethodError: no method matching vec(::Int64)
Closest candidates are:
  vec(::SparseMatrixCSC{Tv,Ti<:Integer}) at sparse/sparsematrix.jl:307
  vec(::AbstractSparseArray{Tv,Ti,1}) at sparse/sparsevector.jl:751
  vec(::AbstractArray{T,1}) at abstractarraymath.jl:36
  ...

julia> convert(Vector, 1)
ERROR: MethodError: Cannot `convert` an object of type Int64 to an object of type Array{T,1}
This may have arisen from a call to the constructor Array{T,1}(...),
since type constructors fall back to convert methods.

julia> collect(1)
0-dimensional Array{Int64,0}:
1

julia> collect([1])
1-element Array{Int64,1}:
 1

julia> [1...]
1-element Array{Int64,1}:
 1

julia> [[1]...]
1-element Array{Int64,1}:
 1
```